### PR TITLE
seed-cache-review-tags-api

### LINKED ISSUE #151: Expose via API endpoint
Update seed cache to load review-tags in `general__memoryStorage`. (`server/utils/tasks/seed-cache.ts`).
Define API endpoint (`server/api/review-tags/index.get.ts`) to read values from cache, authenticate admin request. (see how postings are served)

### PR DESCRIPTION
**Pull request by karthik8239**
## Seed Cache Update and Review-tags api 

- Updated the seed cache to load the review tags from the database schema
- Updated general memory storage to store review tags
- Implements new API endpoint for retrieving review tags
- Secures endpoint with admin authentication
- Fetches review tags from application's memory storage
- Returns array of ReviewTag objects matching database schema
- Enables efficient access to review tags for admin users

Resolved #151

### CODE REVIEW

**Comment by devshahofficial** [comment]
added @amandesai01 as a reviewer!

**Comment by amandesai01** [comment]
link issue.

**Review by amandesai01 — COMMENTED** [review]
Delete all the comments

File: server/utils/tasks/seed-cache.ts [hunk 1]
```diff
@@ -12,6 +12,10 @@ export async function seedCache() {
     .from(metaDataTable)
     .where(inArray(metaDataTable.key, ['seoConfig', 'careerSiteConfig', 'firstSetupAccessKey']));
 
+  //getting the review tags from the reviewTagsTable
+  const reviewTags = await db.select().from(reviewTagsTable);
+  console.log(reviewTags); //debug to check the reviewtags response
```
Comment (amandesai01) [inline server/utils/tasks/seed-cache.ts:17]: delete this

File: server/api/review-tags/index.get.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Import the defineEventHandler function from the 'h3' package
+// This is used to create server-side event handlers
+import { defineEventHandler } from 'h3';
+import { type ReviewTag } from '../../db/schema'; //import review tags from db schema
+
+import authenticateAdminRequest from '../../utils/admin';
```
Comment (amandesai01) [inline server/api/review-tags/index.get.ts:6]: This should be auto-imported

File: server/utils/tasks/seed-cache.ts (hunk 1 above)
Comment (karthik8239) [reply server/utils/tasks/seed-cache.ts:17]: @amandesai01 you mean the comments or code?

File: server/api/review-tags/index.get.ts (hunk 2 above)
Comment (karthik8239) [reply server/api/review-tags/index.get.ts:2]: done

File: server/utils/tasks/seed-cache.ts (hunk 1 above)
Comment (karthik8239) [reply server/utils/tasks/seed-cache.ts]: remove the comments and logging

File: server/api/review-tags/index.get.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { type ReviewTag } from '../../db/schema';
+import authenticateAdminRequest from '../../utils/admin';
```
Comment (amandesai01) [inline server/api/review-tags/index.get.ts:2]:
```suggestion
```

File: server/utils/tasks/seed-cache.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { desc, inArray } from 'drizzle-orm';
-import { jobPostingsTable, metaDataTable } from '~~/server/db/schema';
+import { jobPostingsTable, reviewTagsTable, metaDataTable } from '~~/server/db/schema'; //import the review tags table from schema
```
Comment (amandesai01) [inline server/utils/tasks/seed-cache.ts:2]:
```suggestion
import { jobPostingsTable, reviewTagsTable, metaDataTable } from '~~/server/db/schema';
```

**Review by amandesai01 — CHANGES_REQUESTED** [review]

**Comment by amandesai01** [comment]
@karthik8239 please wait, I am finishing up other tasks. Will take care of this.

**Comment by karthik8239** [comment]
> @karthik8239 please wait, I am finishing up other tasks. Will take care of this.

@amandesai01  Please let me know if I need to I  will complete this

**Pull request merged**